assets/shaders/animate_shader.wgsl

```diff
@@ -1,6 +1,6 @@
-#import bevy_pbr::mesh_types
 // The time since startup data is in the globals binding which is part of the mesh_view_bindings import
-#import bevy_pbr::mesh_view_bindings
+#from bevy_pbr::mesh_view_bindings import globals
+#from bevy_pbr::mesh_vertex_output import MeshVertexOutput
```

Comment:
Is there a particular reason we can't use this syntax?
```
#import bevy_pbr::mesh_vertex_output::MeshVertexOutput
#import bevy_pbr::mesh_vertex_output::{MeshVertexOutput, some_other_thing}
#import bevy_pbr::mesh_vertex_output::*`
````

This is another "non-standard" (or at least, less common) thing that will rub some people the wrong way. We already must have knowledge of "live" module names to resolve imports, so we can decide if something is a module name or a symbol inside of it.

Reply:
> `#import bevy_pbr::mesh_vertex_output::{MeshVertexOutput, some_other_thing}`

this is doable. it would mean braces can't be used in module names, but that's fine.

> `#import bevy_pbr::mesh_vertex_output::MeshVertexOutput`
> `#import bevy_pbr::mesh_vertex_output::*`

not sure about these. we use the imported module names to work out what modules need to be loaded (and to trigger loading them, for path imports). it would break confusingly if modules get loaded out of order (which can definitely happen with async io), or if you misname an imported item and it tells you the module can't be found.

Reply:
I'm not sure I understand what's different about `#import foo::Bar` that makes it more complicated than `#import foo::{Bar, Baz}`

Reply:
>  it would break confusingly if modules get loaded out of order (which can definitely happen with async io), 

This is a good point (and feels like its touching on the core issue).

How do we determine if `#import bevy_pbr::mesh_vertex_output` doesn't exist, or is just still loading?. Currently iirc we rely on `assets.set_untracked` being called ahead of a given load? Am I right in thinking that module names don't work at all (or at least, not reliably) for `server_loaded_shader_asset_file -> server_loaded_shader_asset_file` deps? (both on main and in this pr?).

If so thats a hole that needs filling. This is especially important because we almost certainly want to be able to pre-process shaders at development time (in the short term producing final WGSL, in the long term ideally producing per-platform compiled shaders). This implies resolving these dependencies using the asset system instead of at runtime. I _think_ the new asset system would be able to handle this with "processed ids" (ex: this is how I plan to support UUIDs and other "stable" ids).

However the asset system won't have access to a "global shader identity cache" and it couldn't just listen for things to show up there. It would just be able to wait for a given stable id (such as `bevy_pbr::mesh_vertex_output`) to be resolved (or fail to resolve after all assets have been processed). So that does _probably_ leave us with the need for an unambiguous syntax in this case. We could probably make stable id lookups fallible, but it would mean that shaders with `#import bevy_pbr::mesh_vertex_output::foo` (where foo is a function name) would always be processed "last" and they would be blocking constantly on other work finishing ... not ideal.

So I guess that _does_ probably leave us in the unfortunate situation where module imports must be unambiguous with other symbol imports (unless anyone else has ideas).

In that case, I think my current preferred syntax is in the vein of:

```
// Option A
#import bevy_pbr::mesh_vertex_output MeshVertexOutput
#import bevy_pbr::mesh_vertex_output MeshVertexOutput, some_other_thing
#import bevy_pbr::mesh_vertex_output *

// Option B
#import bevy_pbr:mesh_vertex_output::MeshVertexOutput
#import bevy_pbr:mesh_vertex_output::{MeshVertexOutput, some_other_thing}
#import bevy_pbr:mesh_vertex_output::*

// Option C
#import bevy_pbr/mesh_vertex_output::MeshVertexOutput
#import bevy_pbr/mesh_vertex_output::MeshVertexOutput, some_other_thing}
#import bevy_pbr/mesh_vertex_output::*
```

Not sure how I feel about Option C. It has the interesting (and likely controversial) property of having rough parity with asset paths (with file extensions resolving the ambiguity)

```
#import some/path/file.wgsl::MeshVertexOutput
```

(pretty sure that parity creates confusion but im still sorting out my feelings)

Reply:
I think Option A has my preference

Reply:
I mentioned it a few months ago, but I think we could also consider using `#use` instead of `#import` to mirror rust a bit more closely. There's the downside that people might assume it's exactly like rust of course, but I think it's worth considering.

Reply:
> Am I right in thinking that module names don't work at all (or at least, not reliably) for `server_loaded_shader_asset_file -> server_loaded_shader_asset_file` deps? (both on main and in this pr?)

for manually server-loaded shaders, it works fine in both main and in this pr. in both cases we preprocess the shader to extract the dependencies and wait for them to be loaded ... but this requires that we can identify dependencies without context (and disambiguate them from item imports).

i'm not sure if `AssetPath` (i.e. `#import "module.wgsl"`) shader deps get auto-loaded on main, but they do in this pr.



> ```
> // Option A
> #import bevy_pbr::mesh_vertex_output MeshVertexOutput
> #import bevy_pbr::mesh_vertex_output MeshVertexOutput, some_other_thing
> #import bevy_pbr::mesh_vertex_output *
> ```

i'll give this a go